X3GWriter.py

```diff
@@ -53,7 +79,22 @@ def write(self, stream, nodes, mode = MeshWriter.OutputMode.TextMode):
             if os.path.isfile(binary_filename + "_macos"): #Still fall back to the default name if the MacOS-specific file doesn't exist.
                 binary_filename += "_macos"
 
-        command = [binary_filename, "-p", "-m", "r1d", "-c", os.path.join(binary_path, "cfg.ini"), temp_file, file_name]
+        command = [binary_filename, "-p"]
+
+        container_stack = Application.getInstance().getGlobalContainerStack()
+        if container_stack.getProperty("machine_gcode_flavor", "value") == "Makerbot":
+            command.append("-g")
+        machine = container_stack.getMetaDataEntry("machine_x3g_variant")
```

Comment:
This metadata entry will never exist for any user unless they modified the .global.cfg file by hand with a text editor.

Reply:
Can you elaborate a bit on what you would like to see done here?

If it's making machine_x3g_variant editable in the machine settings dialog in Cura I'd be happy to write a patch for that.

Reply:
So I'm a bit confused. There's no dialog for it, but you don't have to modify global.cfg (I've never even touched that), I just put it in my machine def.json: https://gist.github.com/rpavlik/0fbc5289b2387dfe374a3ab003186845#file-makerbot_replicator_2x-def-json-L21  and it works, at least with 3.2.1...

Reply:
I concur, adding machine_x3g_variant to the machine definition is enough, that's what I did for ZYYX Agile and it works fine on the 3.3 branch.